psi4/src/psi4/libmints/sieve.cc

```diff
@@ -329,5 +370,25 @@ bool ERISieve::shell_significant_qqr(int M, int N, int R, int S) {
     return est >= sieve2_;
 }
 
+bool ERISieve::shell_significant_csam(int M, int N, int R, int S) {
+    // Square of standard Cauchy-Schwarz Q_mu_nu terms (Eq. 1)
+    double mn_mn = shell_pair_values_[N * (size_t)nshell_ + M];
+    double rs_rs = shell_pair_values_[S * (size_t)nshell_ + R];
+
+    // Square of M~_mu_nu terms (Eq. 9)
+    double mm_rr = shell_pair_exchange_values_[R * (size_t)nshell_ + M];
+    double nn_ss = shell_pair_exchange_values_[S * (size_t)nshell_ + N];
+    double mm_ss = shell_pair_exchange_values_[S * (size_t)nshell_ + M];
+    double nn_rr = shell_pair_exchange_values_[R * (size_t)nshell_ + N];
```

Comment:
Would it make sense to define these as ```size_t``` from the start to avoid the casts?

Reply:
Done